feature_server/config/scripts/arena.py

```diff
@@ -260,11 +279,29 @@ def on_team_join(self, team):
                     self.protocol.check_round_end()
             return returned
 
+        def on_position_update(self):
+            if not self.protocol.arena_running:
+                min_distance = None
+                pos = self.world_object.position
+                for spawn in self.team.arena_spawns:
+                    xd = spawn[0] - pos.x
+                    yd = spawn[1] - pos.y
+                    zd = spawn[2] - pos.z
+                    distance = math.sqrt(xd ** 2 + yd ** 2 + zd ** 2)
+                    if min_distance is None or distance < min_distance:
+                        min_distance = distance
+                if min_distance > self.protocol.arena_max_spawn_distance:
+                    self.set_location(random.choice(self.team.arena_spawns))
+                    self.refill()
+            return connection.on_position_update(self)
+
         def get_respawn_time(self):
+            if self.protocol.arena_enabled:
             if self.protocol.arena_running:
```

Comment:
bad indentation should be this:
```python
         def get_respawn_time(self):
            if self.protocol.arena_enabled:
                if self.protocol.arena_running:
                    return -1
                else:
                    return 1
            return connection.get_respawn_time(self);
```

(see https://github.com/NateShoffner/PySnip/pull/6/files#diff-13ae464b925f0aca4965862189c7a1c7R299)